Assets/Scripts/ThirdPersonPlayer.cs

```diff
@@ -60,36 +77,71 @@ public void OnTriggerEnter(Collider collision)
     private void Update()
     {
         stardustCount.text = "Stardust: " + stardust;
-
-        Debug.Log(stardust);
     }
 
     private void FixedUpdate()
     {
+    
+        if (_levelCleared) return;
+        
         /*
         var xAxisInput = Input.GetAxisRaw("Horizontal");
         var yAxisInput = Input.GetAxisRaw("Vertical");
          */
 
         //InputAction replaces "Input.GetAxis("Example")" and calls function
         //movementInput = inputAction.Player.Move.ReadValue<Vector2>();
+        
+        if (_mapActive) return;
+
+        // when on slingshot, make the player move towards the target 
+        if (_onSlingShot)
+        {
+            print("|||||||||| third person player --- slngshot ||||||||||" + _starToGo.name);
+            
+            // sling shot the player to the designated star starToGo
+            var desiredPosition = _starToGo.position + disFromGoalStar;
+            //transform.position = Vector3.MoveTowards(transform.position, desiredPosition, speed * 100 * Time.fixedDeltaTime);
+            transform.position = Vector3.Lerp(transform.position, desiredPosition, speed * 2 * Time.fixedDeltaTime);
+
+            // TODO: disable all the figure when slingshot
+
+            if (Vector3.Distance(desiredPosition, transform.position) < 10.0f)
+            {
+                _onSlingShot = false;
+            }
+        }
+        
         inputAction.Player.Move.performed += ctx => movementInput = ctx.ReadValue<Vector2>();
         inputAction.Player.Move.canceled += ctx => movementInput = Vector2.zero;
 
-        if (_mapActive) return;
-        
+
         var xAxisInput = movementInput.x;
         var yAxisInput = movementInput.y;
+        
+        
+        if (_enableIsometricViewMovement)
+        {
+            // TODO: move the player in a 2D perspective
+            var playerPos = transform.position;
+            playerPos.x += xAxisInput;
+            playerPos.z += xAxisInput;
```

Comment:
If the perspective of the camera is relative to the axis of rotation of the planet,s then playerPos.x and .z may not be correct.

The player's y-direction would first have to be aligned to the star's up vector.